frontend/src/components/LoadingIcon.vue

```diff
@@ -1,5 +1,5 @@
 <template>
-  <div class="lds-ring">
+  <div class="lds-ring" data-testid="lds-ring">
```

Comment:
Should this be using a `role` instead of a `data-testid`? Seems like the icon could be more accessible. See:  https://stackoverflow.com/questions/31716425/how-do-you-make-the-loading-icon-accessible-for-screen-readers-like-jaws

Reply:
I think this should be a different PR, because it has important implications for accessibility.